javascript-data-model/javascript-data-model/main.js

```diff
@@ -0,0 +1,28 @@
+/* exported orderHistory */
+
+var orderHistory = [
+  {
+    customerName: 'John Doe',
+    city: 'Los Angeles',
+    items: 'Bose Headphones',
+    freeGift: true
+  },
+  {
+    customerName: 'Mark Black',
+    city: 'Irvine',
+    items: ['iPad', 'Laptop', 'Computer Science Book', 'Toothbrush'],
+    freeGift: false
+  },
+  {
+    customerName: 'Susan Sunnylife',
+    city: 'San Diego',
+    items: 'Pencils',
+    freeGift: false
+  },
+  {
+    customerName: 'Ted Gilson',
+    city: 'Burbank',
+    items: [{ name: 'Good Cooking', price: 29.99 }, { name: 'Fix Your Car', price: 19.99 }],
```

Comment:
And here is the property `items` with an array of objects. When creating a data model, you want the structure for all items to follow the same pattern. For instance here, having a property that could hold a string, an array of strings, or an array of objects will make it nearly impossible to treat the data for each item in the same way.